### PR TITLE
Update MongoCursor.hx

### PR DESCRIPTION
**Pull request by delahee**
type and document a tiny bit

### CODE REVIEW

File: js/npm/mongodb/MongoCursor.hx [hunk 1]
```diff
@@ -53,4 +53,10 @@ extern class MongoCursor
 	var close               : Dynamic;
 	var isClosed            : Dynamic;
 	
-}
\ No newline at end of file
+	/**
+	End of stream is signaled by a null passed in item
+	*/
+	function each	( func :/*err:*/  Dynamic  -> /*item:*/ Dynamic -> Void ) : Void;
+	function toArray( func :/*err:*/  Dynamic -> /*items:*/ Array<Dynamic> -> Void ) : Void;
```
Comment (dionjwa) [inline js/npm/mongodb/MongoCursor.hx]: It looks like there is a version that returns a value?: https://docs.mongodb.com/manual/reference/method/cursor.toArray/

Comment (delahee) [reply js/npm/mongodb/MongoCursor.hx]: Okay I'll check out if the version I use have the same sig.

**Comment by delahee** [comment]
The returned promise is undefined on some cases :
Promise if no callback passed 

I'll amend and document

**Comment by delahee** [comment]
I added a "Dynamic" as I haven't yet understood how to exploit js.Promise<Any>. Feel free to comment if you know how to do it better :)

File: js/npm/mongodb/MongoCursor.hx [hunk 2]
```diff
@@ -53,4 +53,10 @@ extern class MongoCursor
 	var close               : Dynamic;
 	var isClosed            : Dynamic;
 	
-}
\ No newline at end of file
+	/**
+	End of stream is signaled by a null passed in item
+	*/
+	function each	( func :/*err:*/  Dynamic  -> /*item:*/ Dynamic -> Void ) : Void;
+	function toArray( func :/*err:*/  Dynamic -> /*items:*/ Array<Dynamic> -> Void ) : Dynamic;
```
Comment (dionjwa) [inline js/npm/mongodb/MongoCursor.hx]: I think you can do something like this, when the return (promise or nothing) depends on the arguments:

```
@:overload(function():js.Promise<Array<Dynamic>> { })
function toArray(func :Void<js.Error>->Array<Dynamic>->Void ) : Void;

```

Comment (dionjwa) [reply js/npm/mongodb/MongoCursor.hx]: I'm also honestly fine with the Dynamic case, since maintaining externs is too much human work right now.

**Pull request merged**